pkg/xds/topology/traffictrace.go

```diff
@@ -0,0 +1,26 @@
+package topology
+
+import (
+	"context"
+	core_policy "github.com/Kong/kuma/pkg/core/policy"
+	mesh_core "github.com/Kong/kuma/pkg/core/resources/apis/mesh"
+	core_manager "github.com/Kong/kuma/pkg/core/resources/manager"
+	"github.com/Kong/kuma/pkg/core/resources/store"
+)
+
+func GetTrafficTrace(ctx context.Context, dataplane *mesh_core.DataplaneResource, manager core_manager.ResourceManager) (*mesh_core.TrafficTraceResource, error) {
+	list := mesh_core.TrafficTraceResourceList{}
+	if err := manager.List(ctx, &list, store.ListByMesh(dataplane.GetMeta().GetMesh())); err != nil {
+		return nil, err
+	}
+
+	policies := make([]core_policy.DataplanePolicy, len(list.Items))
+	for i, trace := range list.Items {
+		policies[i] = trace
+	}
+
+	if policy := core_policy.FindBestMatch(dataplane, policies); policy != nil {
```

Comment:
Yeah, I'm testing it and I see that it doesn't support `*` and gateway, but this is scope for other PR. It will be supported once we fix the `FindBestMatch`.